src/zaudio.zig

```diff
@@ -1081,6 +1090,69 @@ pub const decoderReadProc = fn (decoder: *Decoder, buffer_out: *anyopaque, bytes
 pub const decoderSeekProc = fn (decoder: *Decoder, byte_offset: i64, origin: Vfs.SeekOrigin) callconv(.c) Result;
 pub const decoderTellProc = fn (decoder: *Decoder, cursor: *i64) callconv(.c) Result;
 
+//--------------------------------------------------------------------------------------------------
+//
+// Encoder
+//
+//--------------------------------------------------------------------------------------------------
+
+pub const Encoder = opaque {
+    pub const destroy = zaudioEncoderDestroy;
+    extern fn zaudioEncoderDestroy(handle: *Encoder) void;
+
+    pub fn create(encoder_on_write: encoderWriteProc, encoder_on_seek: encoderSeekProc, user_data: *anyopaque, config: Config) Error!*Encoder {
+        var handle: ?*Encoder = null;
+        try maybeError(zaudioEncoderCreate(encoder_on_write, encoder_on_seek, user_data, &config, &handle));
+        return handle.?;
+    }
+    extern fn zaudioEncoderCreate(on_write: encoderWriteProc, on_seek: encoderSeekProc, user_data: *anyopaque, config: *const Config, out_handle: ?*?*Encoder) Result;
+
+    pub fn createFromVfs(vfs: *Vfs, file_path: []const u8, config: Config) Error!*Encoder {
+        var handle: ?*Encoder = null;
+        try maybeError(zaudioEncoderCreateFromVfs(vfs, file_path.ptr, &config, &handle));
+        return handle.?;
+    }
+    extern fn zaudioEncoderCreateFromVfs(vfs: *Vfs, file_path: [*]const u8, config: *const Config, out_handle: ?*?*Encoder) Result;
+
+    pub fn createFromFile(file_path: []const u8, config: Config) Error!*Encoder {
```

Comment:
I think the type of `file_path` should be `[:0]const u8` to force the user to give a sentinel terminated string however they see fit. Note that hardcoded paths should still just work and we don't have to introduce an unwanted copy here. Same applies to `createFromVfs`.